File: elixir/pascals_triangle.ex
defmodule PascalsTriangle do
  @doc """
  Calculates the rows of a pascal triangle
  with the given height
  """
  @spec rows(integer) :: [[integer]]
  def rows(num) do
    num
    |> do_rows()
    |> Enum.reverse()
  end

  defp do_rows(1), do: [[1]]

  defp do_rows(num) do
    rs = do_rows(num - 1)
    prev = hd(rs)

    curr =
      prev
      |> Enum.chunk_every(2, 1)
      |> Enum.flat_map(fn lst -> [Enum.sum(lst)] end)

    [[1 | curr] | rs]
  end
end

File: go/pascals_triangle.go
package pascal

func Triangle(n int) [][]int {
	arr := [][]int{{1}}
	if n == 1 {
		return arr
	}
	for n > 1 {
		n--
		row := []int{1}
		for i := 1; i < len(arr[len(arr)-1]); i++ {
			row = append(row, arr[len(arr)-1][i]+arr[len(arr)-1][i-1])
		}
		row = append(row, 1)
		arr = append(arr, row)
	}
	return arr
}
